src/server/db_slice.cc

```diff
@@ -459,10 +459,11 @@ OpResult<DbSlice::PrimeItAndExp> DbSlice::FindInternal(const Context& cntx, std:
       auto bump_cb = [&](PrimeTable::bucket_iterator bit) {
         DVLOG(2) << "Running callbacks for key " << key << " in dbid " << cntx.db_index;
         for (const auto& ccb : change_cb_) {
-          ccb.second(cntx.db_index, bit);
+          auto cb = ccb;  // we need a copy of shared_ptr to prevent it removing during callback
+          cb->second(cntx.db_index, bit);
         }
```

Comment:
maybe I don't understand something, but modifications while iterating are still unsafe. The vector might re-allocate and all your iterators become invalid (the ones in the for loop)

Reply:
You are right. I also have found one more scenario that can crash. I will fix them tomorrow